orc/src/main/java/org/apache/iceberg/orc/ORC.java

```diff
@@ -64,6 +65,12 @@ private WriteBuilder(OutputFile file) {
       }
     }
 
+    public WriteBuilder forTable(Table table) {
+      schema(table.schema());
+      setAll(table.properties());
```

Comment:
I think this change is unrelated and should go in a separate PR. It's a good idea to add this, just not here.

Also, I think this should do some translation from table properties to ORC properties, like taking `write.orc.compression-codec` and setting the correct property for ORC.

Reply:
Removed.